gobblin-runtime/src/test/java/gobblin/runtime/MultiConverterTest.java

```diff
@@ -225,4 +301,84 @@ public Object convertSchema(Object inputSchema, WorkUnitState workUnit)
       return new EmptyIterable<Object>();
     }
   }
+
+  /**
+   * A {@link Converter} which will alternate between returning a {@link EmptyIterable()},
+   * a {@link SingleRecordIterable()}, or a {@link List()}. The number of records {@link List()} is controlled by the
+   * multiplicity config in the constructor, similar to {@link MultiIdentityConverter()}.
+   */
+  private static class AlternatingConverter extends Converter<Object, Object, Object, Object> {
+
+    private int executionCount = 0;
+
+    private final int multiplicity;
+
+    public AlternatingConverter(int multiplicity) {
+      this.multiplicity = multiplicity;
+    }
+
+    @Override
+    public Object convertSchema(Object inputSchema, WorkUnitState workUnit) throws SchemaConversionException {
+      return inputSchema;
+    }
+
+    @Override
+    public Iterable<Object> convertRecord(Object outputSchema, Object inputRecord, WorkUnitState workUnit)
+        throws DataConversionException {
+
+      if (this.executionCount % 3 == 0) {
```

Comment:
You can increment `executionCount` first, and then check for `this.executionCount % 1`, `this.executionCount % 2`, and `this.executionCount % 3`, respectively. This guarantees it returns iterators in a consistent order.